substrate/frame/contracts/src/migration.rs

```diff
@@ -240,13 +242,14 @@ impl<T: Config, const TEST_ALL_STEPS: bool> Migration<T, TEST_ALL_STEPS> {
 		loop {
 			let in_progress_version = <Pallet<T>>::on_chain_storage_version() + 1;
 			let state = T::Migrations::pre_upgrade_step(in_progress_version)?;
-			let (status, w) = Self::migrate(Weight::MAX);
-			weight.saturating_accrue(w);
+			let mut meter = &mut WeightMeter::new();
+			let status = Self::migrate(&mut meter);
+			weight.saturating_accrue(meter.consumed());
```

Comment:
Is this still needed? Can't we use `meter.weight_consumed()` everywhere?

Reply:
Yep this was just to reflect the weight of a single migrate call, but we can do it as well as you suggest.